edsdme/blocks/yukon-chat/yukon-chat.js

```diff
@@ -0,0 +1,176 @@
+import { getCurrentProgramType, getPartnerDataCookieObject, partnerIsSignedIn, aemPublish } from '../../scripts/utils.js';
+
+export default async function init(el) {
+  const app = document.createElement('div');
+  app.className = 'yukon-chat';
+  const chatDialog = document.createElement('div');
+  chatDialog.className = 'chat-dialog';
+  const dialogBody = document.createElement('div');
+  dialogBody.className = 'yukon-chat-dialog-body';
+  const title = document.createElement('h4');
+  title.className = 'yukon-chat-title';
+  title.textContent = 'Yukon AI Chat';
+  const divider = document.createElement('hr');
+  const description = document.createElement('span');
+  description.className = 'yukon-chat-description';
+  description.textContent = 'Ask your question:';
+  const textarea = document.createElement('textarea');
+  textarea.className = 'yukon-chat-textarea';
+  textarea.placeholder = 'Type your message here...';
+  textarea.rows = 3;
+  const responseLabel = document.createElement('span');
+  responseLabel.className = 'yukon-chat-response-label';
+  responseLabel.textContent = 'Response:';
+  const responseArea = document.createElement('textarea');
+  responseArea.className = 'yukon-chat-response';
+  responseArea.placeholder = 'Response will appear here...';
+  responseArea.rows = 5;
+  responseArea.readOnly = true;
+
+  const referencesLabel = document.createElement('span');
+  referencesLabel.className = 'yukon-chat-references-label';
+  referencesLabel.textContent = 'References:';
+  referencesLabel.style.display = 'none';
+
+  const referencesContainer = document.createElement('div');
+  referencesContainer.className = 'yukon-chat-references';
+  referencesContainer.style.display = 'none';
+
+  const sendButton = document.createElement('button');
+  sendButton.className = 'yukon-chat-button cta';
+  sendButton.textContent = 'Send';
+
+  const requestId = crypto.randomUUID();
+  sendButton.addEventListener('click', async () => {
+    const question = textarea.value.trim();
+    if (!question) return;
+    responseArea.value = 'Loading...';
+    referencesLabel.style.display = 'none';
+    referencesContainer.style.display = 'none';
+    referencesContainer.innerHTML = '';
+    let level = 'public';
+    let region = 'worldwide';
+    if (partnerIsSignedIn()) {
+      try {
+        const profileData = getPartnerDataCookieObject(getCurrentProgramType());
```

Comment:
please add a TODO: the partner data must be sent to the servlet and parsed there